ocp support on m1 post (#5)
code-ready/blog: content/posts/2022-08-17-ocp-on-m1.md

---
title: "Running ocp on M1 machines"
date: 2022-08-17T08:58:10+05:30
---

> From the 2.4.1 release, CRC has been running on macOS aarch64 M1 CPUs.

Reviewer: With the release of CRC 2.4.1 it was possible to run on Apple's latest offering of M1-based machines.

> However, the support was limited to podman preset on M1. We've been asked numerous times when CRC openshift

Reviewer: ... remove M1 if the previous suggestion is accepted.

> preset would get M1 support. This has changed with the recently released [crc 2.7.1](https://github.com/code-ready/crc/releases/tag/v2.7.1).

Reviewer: Numerous times we received requests if it would be possible to instead offer an OpenShift that would run on these machines.

This blog post will describe the challenges we had to overcome when we decided to add that support.

# Background

> CRC is a tool which allows you to run an OpenShift cluster or podman containers on a laptop or local workstation.

Reviewer: CRC is a tool that manages the lifecycle of a virtual machine ...

It can be used on a macOS, Windows or Linux machine.

> It runs its OpenShift cluster/podman runtime by starting a virtual machine (VM) using a disk image built in advance and packaged in what we call a _bundle_.

Reviewer: remove 'by starting a virtual machine (VM) '

> On each platform, we try to use native hypervisors.

Reviewer: , such as KVM on Linux, Hyper-V on Windows, and using the virtualization framework on macOS.

# Issue of running OCP on M1

Openshift container platform (OCP) uses Red Hat Core OS (RHCOS) which is based on latest RHEL-8 which is

> compiled for a 64K page size (aarch64), but Apple M1 processor uses 16K or 4K page sizes [link](https://access.redhat.com/discussions/5966451). 

Reviewer: uses 4K or a maximum size of 16K,

Reviewer: not sure if '(aarch64)' here is relevant, as I believe that is also the case for x86 IIRC. Otherwise write 'for aarch64 targets'.

> # How it now works

Reviewer: 'now' ?

> Now, after we provision the Openshift cluster we upgrade the RHEL-8 kernel packages to RHEL-9 which doesn't have issue to boot on

Reviewer: remove 'Now, '.

'upgraded' or better 'replaced'

> compiled for a 64K page size, but Apple M1 processor uses 4K or a maximum size of 16K, page sizes [link](https://access.redhat.com/discussions/5966451). 

Reviewer: `16K, page sizes` Why the `,`

> compiled for a 64K page size, but Apple M1 processor uses 4K or a maximum size of 16K, page sizes [link](https://access.redhat.com/discussions/5966451). 

Reviewer 2: ```suggestion
compiled for a 64K page size, but Apple M1 processor uses 4K or a maximum size of 16K page sizes [link](https://access.redhat.com/discussions/5966451). 
```

> Apple M1 machines. As of now this workaround fix current issue, and we are able to run the created disk image on M1.

Reviewer 2: ```suggestion
Apple M1 machines. As of now this workaround fixes current issue, and we are able to run the created disk image on M1.
```

> Apple M1 machines. As of now this workaround fixes current issue, and we are able to run the created disk image on M1.

Reviewer: Using this kernel it is possible to workaround this issue and we are able to ...

```shell
$ rpm -qa | grep kernel
kernel-5.14.0-70.13.1.el9_0.aarch64
kernel-core-5.14.0-70.13.1.el9_0.aarch64
kernel-modules-extra-5.14.0-70.13.1.el9_0.aarch64
kernel-modules-5.14.0-70.13.1.el9_0.aarch64
```

# Where we're at

> Once RHCOS switches to RHEL-9, we don't need to upgrade the kernel bits manually and remove this workaround.

Reviewer 2: ```suggestion
Once RHCOS switches to RHEL-9, we don't need to upgrade the kernel bits manually and this workaround can be removed.
```

> Once RHCOS switches to RHEL-9, we don't need to upgrade the kernel bits manually and this workaround can be removed.

Reviewer: replace
